src/tools/recipe2plan.ts

```diff
@@ -12,24 +12,36 @@ import {StorageKeyRecipeResolver} from './storage-key-recipe-resolver.js';
 import {PlanGenerator} from './plan-generator.js';
 import {Flags} from '../runtime/flags.js';
 import {assert} from '../platform/assert-node.js';
+import {encodePlansToProto} from './manifest2proto.js';
 
+export enum OutputFormat { Kotlin, Proto }
 
 /**
  * Generates Kotlin Plans from recipes in an arcs manifest.
  *
  * @param path path/to/manifest.arcs
  * @return Generated Kotlin code.
  */
-export async function recipe2plan(path: string): Promise<string> {
+export async function recipe2plan(
+    path: string,
+    format: OutputFormat,
+    recipeFilter?: string): Promise<string | Uint8Array> {
   return await Flags.withDefaultReferenceMode(async () => {
     const manifest = await Runtime.parseFile(path);
+    let plans = await (new StorageKeyRecipeResolver(manifest)).resolve();
 
-    assert(manifest.meta.namespace, `Namespace is required in '${path}' for code generation.`);
+    if (recipeFilter) {
+      plans = plans.filter(p => p.name === recipeFilter);
```

Comment:
Could there be an issue where more than one recipe is found? 
Maybe use `find` which returns the first element only, or `undefined`.

Reply:
After seeing the code below, this seems to be handled gracefully. So, consider this an issue optionally.